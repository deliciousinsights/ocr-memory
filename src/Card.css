.memory > .card {
  font-size: 2em;
  flex: 1 1 calc(100% / 6 - 0.4em);
  outline: 0.08em solid silver;
  margin: 0.2em;
  display: flex;
  cursor: default;
}

.memory > .card.hidden {
  background: silver;
}

.memory > .card.justMatched,
.memory > .card.justMismatched {
  outline: 0.1em solid green;
}
.memory > .card.justMismatched {
  outline-color: red;
}

.memory > .card.visible {
  cursor: not-allowed;
}

.memory > .card > .symbol {
  margin: auto;
}
