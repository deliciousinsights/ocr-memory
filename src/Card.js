import PropTypes from 'prop-types'
import React from 'react'

import './Card.css'

const HIDDEN_SYMBOL = '❓'

const Card = ({ card, feedback, index, onClick }) => (
  <div className={`card ${feedback}`} onClick={() => onClick(index)}>
    <span className="symbol">
      {feedback === 'hidden' ? HIDDEN_SYMBOL : card}
    </span>
  </div>
)

Card.propTypes = {
  card: PropTypes.string.isRequired,
  feedback: PropTypes.oneOf([
    'hidden',
    'justMatched',
    'justMismatched',
    'visible',
  ]).isRequired,
  index: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Card
