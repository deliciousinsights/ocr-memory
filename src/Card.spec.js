import { expect } from 'chai'
import React from 'react'
import { shallow } from 'enzyme'
import sinon from 'sinon'

import Card from './Card'

describe('<Card/>', () => {
  it('should trigger its `onClick` prop when clicked', () => {
    const onClick = sinon.spy()
    const wrapper = shallow(
      <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
    )

    wrapper.simulate('click')
    expect(onClick).to.have.been.calledWith(0)
  })

  it('should match its reference snapshot', () => {
    const onClick = sinon.spy()
    const wrapper = shallow(
      <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
    )

    expect(wrapper).to.matchSnapshot()
  })
})
