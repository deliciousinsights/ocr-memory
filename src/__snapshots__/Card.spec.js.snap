// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should match its reference snapshot 1`] = `
<div
  className="card hidden"
  onClick={[Function]}
>
  <span
    className="symbol"
  >
    ‚ùì
  </span>
</div>
`;
